### ISSUE TITLE
Change approach

### ISSUE THREAD

**Issue by Shulepov** [krzyzanowskim/Natalie#27]
Hi,
I want to suggest to slightly change the approach of what should be generated.
Currently generated extensions for view controllers that overrides some property.

But it has one drawback:
- only one view controller in inheritance hierarchy can have storyboard identifier, or generated file won't compile (as extensions can override properties and methods only once, e.g. if we have two view controllers _BaseViewController_ and _ExtendedViewController: BaseViewController_, that both have storyboard identifiers);

So, instead of generating extensions for view controllers can be generated entities for storyboard identifiers, e.g.:

``` swift
protocol StoryboardIdentifier {
    var name: String { get }
}

protocol ViewControllerIdentifier: StoryboardIdentifier {
    func instantiateViewControllerFromStoryboard(storyboard: Storyboards) -> UIViewController! 
}
```

And for each identifier in storyboard generate class:

``` swift
final class MenuViewControllerIdentifier: ViewControllerIdentifier {
    var name: String { return "MenuViewController" }
    func instantiateViewControllerFromStoryboard(storyboard: Storyboards) -> MenuViewController! {
         return storyboard.instantiateViewControllerWithIdentifier(self.name) as? MenuViewController
    }
}
```

Such approach implies to better name identifiers in storyboards.

Or second solution, something like this (similar whats done for segues):

``` swift
extension MenuViewController {
     class StoryboardEntry {
            case MainMenu("MainMenu")
            case SecondaryMenu("SecondaryMenu")
            func instantiateFromStoryboard(storyboard: Storyboards) -> MenuViewController! {
                   return storyboard.instantiateViewControllerWithIdentifier(self.rawValue) as? MenuViewController
            }
     }
}
```

P.S. Possibly better names can be given to the protocols and generated classes.

**Comment by krzyzanowskim** [comment]
This was addressed with #32 by using nested types in place of enum. I believe this solve the issues mentioned here. @Shulepov can you please confirm ?

**Comment by Shulepov** [comment]
Yes, thanks, seems like #32 solves the issue in a very good way :)
